sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImplTest.java

```diff
@@ -908,6 +909,53 @@ public void testInsertWithinRowCountLimits() throws Exception {
     verifyWriteMetricWasSet("project", "dataset", "table", "ok", 3);
   }
 
+  /** Tests that {@link DatasetServiceImpl#insertAll} does not go over limit of rows per request. */
+  @Test(expected = RuntimeException.class)
+  public void testInsertWithinRequestByteSizeLimitsErrorsOut() throws Exception {
+    TableReference ref =
+        new TableReference().setProjectId("project").setDatasetId("dataset").setTableId("table");
```

Comment:
Can we also add a test for when retry policy is "retryTransientErrors" ?

Reply:
done